src/analysis/multiSpecies/microbiomeModelingToolbox/mgPipe/microbiotaModelSimulator.m

```diff
@@ -0,0 +1,242 @@
+  if extSolve==0
+      AllRxn = model.rxns;
+      FecalInd  = find(cellfun(@(x) ~isempty(strfind(x,'[fe]')),AllRxn));
+      DietInd  = find(cellfun(@(x) ~isempty(strfind(x,'[d]')),AllRxn));
+      FecalRxn = AllRxn(FecalInd);
+      FecalRxn=setdiff(FecalRxn,'EX_microbeBiomass[fe]','stable');
+      DietRxn = AllRxn(DietInd);
+       if rDiet==1
+          [minFlux,maxFlux]=guidedSim(model,fvaType,FecalRxn);
+           sma=maxFlux;
+           sma2=minFlux
+          [minFlux,maxFlux]=guidedSim(model,fvaType,DietRxn);
+          smi=minFlux;
+          smi2=maxFlux
+          maxFlux=sma;
+          minFlux=smi;
+          fvaCt{1,(k-1)}=ID;
+          nsCt{1,(k-1)}=ID;
+          for i =1:length(FecalRxn)
+            [truefalse, index] = ismember(FecalRxn(i), ID);
+            fvaCt{1,(k-1)}{index,2}=minFlux(i,1);
+            fvaCt{1,(k-1)}{index,3}=maxFlux(i,1);
+            nsCt{1,(k-1)}{index,2}=smi2(i,1);
+            nsCt{1,(k-1)}{index,3}=sma2(i,1);
+          end
+     end
+  else
+       microbiota_model=model
+       mkdir(strcat(resPath,'Rich'))
+       save(strcat(resPath,'Rich\','microbiota_model_richD_',idInfo,'.mat'),'microbiota_model')
+  end
+
+
+% Using standard diet
+
+model_sd=model;
+[adaptedDiet] = adaptVMHDietToAGORA(dietFilePath,'Microbiota')
+[model_sd] = useDiet(model_sd, adaptedDiet)
+    if exist('unfre') ==1 %option to directly add other essential nutrients
+       warning('Feasibility forced with addition of essential nutrients')
+       model_sd=changeRxnBounds(model_sd, unfre,-0.1,'l')
+    end
+solution_sDiet=solveCobraLP(model_sd);
+% solution_sDiet=solveCobraLPCPLEX(model_sd,2,0,0,[],0);
+presol{k,2}=solution_sDiet.obj
+ if isnan(solution_sDiet.obj)
+    warning('Presolve detected one or more infeasible models. Please check InFesMat object !')
+    inFesMat{k,2}= model.name
+ else
+
+  if extSolve==0
+      [minFlux,maxFlux]=guidedSim(model_sd,fvaType,FecalRxn);
+       sma=maxFlux;
+       sma2=minFlux
+       [minFlux,maxFlux]=guidedSim(model_sd,fvaType,DietRxn);
+       smi=minFlux;
+       smi2=maxFlux
+       maxFlux=sma;
+       minFlux=smi;
+
+       fvaCt{2,(k-1)}=ID;
+       nsCt{2,(k-1)}=ID;
+        for i =1:length(FecalRxn)
+            [truefalse, index] = ismember(FecalRxn(i), ID);
+            fvaCt{2,(k-1)}{index,2}=minFlux(i,1);
+            fvaCt{2,(k-1)}{index,3}=maxFlux(i,1);
+            nsCt{2,(k-1)}{index,2}=smi2(i,1);
+            nsCt{2,(k-1)}{index,3}=sma2(i,1);
+        end
+  else
+  microbiota_model=model_sd;
+  mkdir(strcat(resPath,'Standard'))
+  save(strcat(resPath,'Standard\','microbiota_model_standardD_',idInfo,'.mat'),'microbiota_model')
+  end
+
+if extSolve==0
+   save(strcat(resPath,'intRes.mat'),'fvaCt','presol','inFesMat', 'nsCt')
+
+end
+
+
+% Using personalized diet not documented in MgPipe and bug checked yet!!!!
+
+if pDiet==1
+  model_pd=model;
+ [Numbers, Strings] = xlsread(strcat(abundancepath,fileNameDiets));
```

Comment:
I'd suggest using a unix/winodws switch here. and use xlread in case its a unix system.